### PR TITLE
[feat] custom exchange providers

### PR DESCRIPTION
**Pull request by motorina0**
### Summary
- allow admins to customize their exchange providers
- show chart for recent exchange data (in order to easily identify anomalies)

image

### CODE REVIEW

File: lnbits/core/tasks.py [hunk 1]
```diff
@@ -188,3 +189,27 @@ async def purge_audit_data():
...
+                rates = await btc_rates(currency)
+                if rates:
+                    rates_values = [r[1] for r in rates]
+                    lnbits_rate = sum(rates_values) / len(rates_values)
+                    rates.append(("LNbits", lnbits_rate))
+                settings.append_exchange_rate_datapoint(dict(rates), max_history_size)
+            except Exception as ex:
+                logger.warning(ex)
+        else:
+            sleep_time = 60
```
Comment (dni) [inline lnbits/core/tasks.py:214]:
```suggestion
        sleep_time = settings.lnbits_exchange_history_refresh_interval_seconds
        sleep_time = sleep_time if sleep_time > 0 else 60

        try:
                rates = await btc_rates(currency)
                if rates:
                rates_values = [r[1] for r in rates]
                lnbits_rate = sum(rates_values) / len(rates_values)
                rates.append(("LNbits", lnbits_rate))
            settings.append_exchange_rate_datapoint(dict(rates), max_history_size)
        except Exception as ex:
            logger.warning(ex)
```

Comment (motorina0) [reply lnbits/core/tasks.py:214]: if `sleep_time` is zero or negative it should not try to fetch the exchange rates

File: lnbits/settings.py [hunk 2]
```diff
@@ -250,6 +272,71 @@ def fee_reserve(self, amount_msat: int, internal: bool = False) -> int:
...
 
 
+class ExchangeProvidersSettings(LNbitsSettings):
+    lnbits_exchange_history_size: int = Field(default=60)
+    lnbits_exchange_history_refresh_interval_seconds: int = Field(default=300)
+
+    lnbits_exchange_rate_providers: list[ExchangeRateProvider] = Field(
+        default=[
+            ExchangeRateProvider(
+                name="Binance",  # todo: USDT instead of USD
```
Comment (dni) [inline lnbits/settings.py:282]: remove

File: lnbits/settings.py [hunk 3]
```diff
@@ -250,6 +272,71 @@ def fee_reserve(self, amount_msat: int, internal: bool = False) -> int:
...
+                path="$.data.last",
+            ),
+            ExchangeRateProvider(
+                name="Kraken",
+                api_url="https://api.kraken.com/0/public/Ticker?pair=XBT{TO}",
+                path="$.result.XXBTZ{TO}.c[0]",
+                exclude_to=["czk"],
+            ),
+            ExchangeRateProvider(
+                name="BitPay",
```
Comment (dni) [inline lnbits/settings.py:327]: remove

**Review by dni — COMMENTED** [review]
you have a file ` lnbits/static/bundle-components.min.js.old` in there

**Pull request merged**